File: pkg/splprime/ast/statements/block.go
package statements

const DEFAULT_CAPACITY = 8

type Block struct {
	children []Statement
}

func NewBlock() *Block {
	return &Block{
		children: make([]Statement, 0, DEFAULT_CAPACITY),
	}
}

func NewBlockWith(children ...Statement) *Block {
	return &Block{
		children: children,
	}
}

func (s *Block) Add(st Statement) {
	s.children = append(s.children, st)
}

func (s *Block) Children() []Statement {
	return s.children
}

func (s *Block) Accept(v Visitor) {
	for _, child := range s.children {
		child.Accept(v)
	}
}

File: src/splprime/ast/statements/Block.java
package splprime.ast.statements;

import java.util.ArrayList;
import java.util.Arrays;

import splprime.visitor.StatementVisitor;

public class Block extends Statement {

    private final int DEFAULT_CAPACITY = 8;
    ArrayList<Statement> children;

    public Block() {
        this.children = new ArrayList<>(DEFAULT_CAPACITY);
    }

    public Block(Statement... children) {
        this.children = new ArrayList<>(Arrays.asList(children));
    }

    public void add(Statement statement) {
        this.children.add(statement);
    }

    public Statement[] getChildren() {
        return (Statement[]) children.toArray();
    }

    @Override
    public void accept(StatementVisitor v) {
        for (var child : children) {
            child.accept(v);
        }
    }
}
